### PR TITLE
Allow Cylc VIP to work with named runs.

### LINKED ISSUE #5310: VIP: Run Name not handled
**Describe the bug**

If you use the `--run-name` option with `cylc install` you have to specify the run name in the ID with `cylc play`, unlike the numbered runs where cylc assumes you mean the most recent install:

```bash
cd ~/cylc-src/foo
cylc vip --run-name bar
...
# validate and install OK
...
WorkflowFilesError: No flow.cylc or suite.rc in /home/h02/tpilling/cylc-run/foo  # It's in foo/bar
```

**Expected behavior**

Either
1. Throw `--run-name` out from Cylc VIP as an invalid option.
2. Allow `--run-name` and modify the workflow id given to Cylc play accordingly.
3. Modify `cylc install` and `cylc play` so that `cylc install` can mark the last install. `cylc play` uses marker to infer run-name to use. 

**Additional context**

It might be worth checking various permutations of run-name & no-run-name looking for other bugs like this one.

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

### PR DESCRIPTION
**Pull request by wxtim**
Closes #5310 

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users

### CODE REVIEW

**Review by datamel — COMMENTED** [review]
I think the tests need checking on this one.

File: cylc/flow/workflow_files.py [hunk 1]
```diff
@@ -1737,7 +1737,7 @@ def install_workflow(
     install_log.info(f'INSTALLED {named_run} from {source}')
     print(f'INSTALLED {named_run} from {source}')
     close_log(install_log)
-    return source, rundir, workflow_name
+    return source, rundir, named_run
```
[the remark this answers is not in the document]
Comment (wxtim) [reply cylc/flow/workflow_files.py:1740]: No, I think I've broken the functionality of Cylc install sadly. Will have a wee think.

File: tests/functional/cylc-combination-scripts/06-vip-named-run.t [hunk 2]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (wxtim) [reply tests/functional/cylc-combination-scripts/06-vip-named-run.t:1]: Still?

**Comment by oliver-sanders** [comment]
```
$ cylc install generic --run-name foo
INSTALLED generic/foo from ~/cylc-src/generic

$ cylc vip generic --run-name bar
$ cylc validate ~/cylc-src/generic
Valid for cylc-8.1.0
$ cylc install ~/cylc-src/generic
INSTALLED generic/bar from ~/cylc-src/generic
$ cylc play generic

 ▪ ■  Cylc Workflow Engine 8.1.0
 ██   Copyright (C) 2008-2023 NIWA
▝▘    & British Crown (Met Office) & Contributors

WorkflowFilesError: No flow.cylc or suite.rc in ~/cylc-run/generic
```

**Comment by oliver-sanders** [comment]
Could you rebase onto 8.1.x please.

**Comment by wxtim** [comment]
> ```
> $ cylc install generic --run-name foo
> INSTALLED generic/foo from ~/cylc-src/generic
> 
> $ cylc vip generic --run-name bar
> $ cylc validate ~/cylc-src/generic
> Valid for cylc-8.1.0
> $ cylc install ~/cylc-src/generic
> INSTALLED generic/bar from ~/cylc-src/generic
> $ cylc play generic
> 
>  ▪ ■  Cylc Workflow Engine 8.1.0
>  ██   Copyright (C) 2008-2023 NIWA
> ▝▘    & British Crown (Met Office) & Contributors
> 
> WorkflowFilesError: No flow.cylc or suite.rc in ~/cylc-run/generic
> ```

I haven't been able to replicate this.

File: cylc/flow/workflow_files.py [hunk 3]
```diff
@@ -1633,7 +1633,7 @@ def install_workflow(
     run_name: Optional[str] = None,
     no_run_name: bool = False,
     cli_symlink_dirs: Optional[Dict[str, Dict[str, Any]]] = None
-) -> Tuple[Path, Path, str]:
+) -> Tuple[Path, Path, str, str]:
```
Comment (MetRonnie) [inline cylc/flow/workflow_files.py:1636]: Docstring return value needs updating

File: CHANGES.md [hunk 4]
```diff
@@ -11,6 +11,14 @@ updated. Only the first match gets replaced, so it's fine to leave the old
...
 
 -------------------------------------------------------------------------------
+## __cylc-8.1.1 (<span actions:bind='release-date'>Coming Soon</span>)__
+
+### Fixes
+
+[#5314](https://github.com/cylc/cylc-flow/pull/5314) - Fix broken
+command option: `cylc vip --run-name`.
+
+
```
Comment (MetRonnie) [inline CHANGES.md:21]:
```suggestion
-------------------------------------------------------------------------------
```

**Review by MetRonnie — COMMENTED** [review]
I cannot reproduce Oliver's "No flow.cylc or suite.rc in ~/cylc-run/generic"

File: tests/functional/cylc-combination-scripts/06-vip-named-run [hunk 5]
```diff
@@ -0,0 +1 @@
+00-vip
```
[the remark this answers is not in the document]
Comment (wxtim) [reply tests/functional/cylc-combination-scripts/06-vip-named-run:1]: It's a symlink - it uses the same workflow as 00-vip

File: cylc/flow/workflow_files.py (hunk 3 above)
Comment (wxtim) [reply cylc/flow/workflow_files.py:1636]: DOne

**Comment by wxtim** [comment]
> Will this need a cylc-rose PR?

I don't think cylc-run uses the changed interfaces, and running the tests suggests that it doesn't.

File: tests/functional/cylc-combination-scripts/06-vip-named-run.t [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+cp -r "${TEST_SOURCE_DIR}/${TEST_NAME_BASE}/flow.cylc" .
+cp -r "${TEST_SOURCE_DIR}/${TEST_NAME_BASE}/reference.log" .
+
+run_ok "${TEST_NAME_BASE}-from-path" \
+    cylc vip --no-detach --debug \
+    --workflow-name "${WORKFLOW_NAME}" \
+    --initial-cycle-point=1300 \
+    --run-name sardine \
+    --reference-test \
```
Comment (oliver-sanders) [inline tests/functional/cylc-combination-scripts/06-vip-named-run.t:34]:
```suggestion
    --reference-test
```

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**